types/k6/experimental/browser.d.ts

```diff
@@ -2570,7 +2570,7 @@ export interface Page {
      * `https://`.
      * @param options
      */
-    goto(url: string, options?: NavigationOptions): Promise<null | Response>;
+    goto(url: string, options?: NavigationOptions): Promise<Response | null>;
```

Comment:
This ordering makes more sense. Is this also idiomatic?

Reply:
I would say so, yes.

My gut feeling for how it "should" be is:

1. Complex, named types
2. Primitives
3. `null` and `undefined`

For example:

```ts
function abc(): Alphabet | string | null
```

The point is to keep the thing that the user is looking for up front, so that it's easier to find functions that return what you want. And you're unlikely to want to use a function because it returns `null` or `undefined`.

Reply:
The counter-argument is that you should draw attention to the exceptional value (`null`, `undefined` or custom error types) to highlight the fact that the function has some special cases that you need to handle.

Reply:
But what I see most often is that `null` and `undefined` go at the end. 🙂